src/jailer/src/main.rs

```diff
@@ -235,21 +244,71 @@ pub fn readln_special<T: AsRef<Path>>(file_path: &T) -> Result<String> {
     Ok(line)
 }
 
-fn sanitize_process() {
-    // First thing to do is make sure we don't keep any inherited FDs
-    // other that IN, OUT and ERR.
-    // SAFETY: Always safe.
-    let fd_limit = i32::try_from(unsafe { libc::sysconf(libc::_SC_OPEN_MAX) }).unwrap();
-    // Close all file descriptors excluding 0 (STDIN), 1 (STDOUT) and 2 (STDERR).
-    for fd in 3..fd_limit {
-        // SAFETY: Safe because close() cannot fail when passed a valid parameter.
-        unsafe {
-            libc::close(fd);
+fn close_fds_by_close_range() -> Result<()> {
+    // First try using the close_range syscall to close all open FDs in the range of 3..UINT_MAX
+    // SAFETY: if the syscall is not available then ENOSYS will be returned
+    SyscallReturnCode(unsafe {
+        libc::syscall(
+            libc::SYS_close_range,
+            3,
+            libc::c_uint::MAX,
+            libc::CLOSE_RANGE_UNSHARE,
+        )
+    } as libc::c_int)
+    .into_empty_result()
+    .map_err(Error::CloseRange)
```

Comment:
Nit: do we need this dedicated error since we discard the error when this call fails and we have a fallback mechanism?

Reply:
That makes sense. Let's keep it as is.